akka-http-core/src/main/java/akka/http/javadsl/model/headers/SameSite.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.javadsl.model.headers;
+
+/**
+ * The Cookie SameSite attribute as defined by <a href="https://tools.ietf.org/html/draft-ietf-httpbis-cookie-same-site-00">RFC6265bis</a>.
+ */
+public enum SameSite {
+    Strict, Lax, None
```

Comment:
Having an enum is nice. Do we need to provide an escape hatch for the future when other things might be in there?

Do we need `None` since it will be packed into an Option? What would `Some(None)` mean as a value opposed to `None`?

Reply:
Looking into the spec, the [grammar](https://tools.ietf.org/html/draft-ietf-httpbis-cookie-same-site-00#section-3.1) only seems to allow `Strict` and `Lax`.

IIUC, the part about a `None` value is about how a browser should store such an attribute which needs to store the fact that the attribute was missing. We already have the `Option` type for that, so I guess `None` wouldn't be needed as part of the enum here. Or am I reading things wrong?